scipy/stats/_continuous_distns.py

```diff
@@ -9628,14 +9630,13 @@ def solve_for_kappa(kappa):
                                    bracket=(1e-8, 1e12))
             return root_res.root
 
-        if floc is None:
-            floc = find_mu(data)
-            fshape = find_kappa(data)
-        else:
-            fshape = find_kappa(data)
+        # location likelihood equation has a solution independent of kappa
+        loc = floc if floc is not None else find_mu(data)
+        # shape likelihood equation depends on location
+        shape = fshape if fshape is not None else find_kappa(data, loc)
```

Comment:
This also allows the user to fix the shape.

Note that it is preferable to call the final values of the location and shape `loc` and `shape` rather than `floc` and `fshape`, since they were not actually `f`ixed.